clients/tfchain-client-go/impl.go

```diff
@@ -37,145 +36,437 @@ type Versioned struct {
 type Conn = *gsrpc.SubstrateAPI
 type Meta = *types.Metadata
 
+// Pool connection
+type poolConn struct {
+	conn     Conn
+	meta     Meta
+	url      string
+	lastUsed atomic.Int64 // Unix timestamp
```

Comment:
I don't believe the poolConn is used anywhere without the mutex, so probably better to get rid of all of the atomics.